### PR TITLE
Add Treasure Hunt

### PR DESCRIPTION
**Pull request by techmaharaj**
Adding example for a treasure hunt game.

### CODE REVIEW

**Comment by techmaharaj** [comment]
@cloudnautique can you review this?

**Comment by techmaharaj** [comment]
@cloudnautique can you please review this?

File: examples/treasure-hunt/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+### Steps
+
+1. Clone the repository:
+
+    ``` bash
+    git clone https://github.com/gptscript-ai/gptscript.git
+    ```
+
+2. Navigate to the `examples/treasure-hun` directory and install the dependencies:
```
Comment (cloudnautique) [inline examples/treasure-hunt/README.md:24]: typo, missing the 't' at the end of hunt.

File: examples/treasure-hunt/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Flask==2.0.1
```
Comment (cloudnautique) [inline examples/treasure-hunt/requirements.txt:1]: Pin a version of werkzeug or use a newer Flask. The newest werkzeug is incompatible with this version of Flask.

File: examples/treasure-hunt/README.md (hunk 1 above)
Comment (techmaharaj) [reply examples/treasure-hunt/README.md]: Done.

File: examples/treasure-hunt/requirements.txt (hunk 2 above)
Comment (techmaharaj) [reply examples/treasure-hunt/requirements.txt:1]: Added.

**Comment by cloudnautique** [comment]
If we are going to host this on Render, can we add a Docker container? Similar to how the gptscript-ai/story-book app is done? Also, the file is hardcoded, so concurrent users would stomp on each other, would be good if that were dynamic.

It doesn't look like you are using any external tools, so you wouldn't need to run gptscript --assemble to aggregate everything at build time.

**Comment by techmaharaj** [comment]
> If we are going to host this on Render, can we add a Docker container? Similar to how the gptscript-ai/story-book app is done? Also, the file is hardcoded, so concurrent users would stomp on each other, would be good if that were dynamic.
> 
> It doesn't look like you are using any external tools, so you wouldn't need to run gptscript --assemble to aggregate everything at build time.

Hi @cloudnautique this is done, please have a look.

File: examples/treasure-hunt/templates/index.html [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            <button class="button is-primary is-medium" style="background-color: #140aa2;" @click="getClues" :disabled="isLoading">
+                <span v-if="isLoading" class="icon is-medium">
+                    <i class="fas fa-spinner fa-spin"></i>
+                </span>
+                <span v-else>Get Clues</span>
+            </button>
+        </div>
+
+        <div class="box markdown-body" v-if="showClues" v-html="renderedMarkdown"></div>
+        <div class="gpt-logo"><img src="{{url_for('static', filename='made-with-gptscript.png')}}"/></div>
```
Comment (cloudnautique) [inline examples/treasure-hunt/templates/index.html:38]: Should this link to gptscript.ai or something?

File: examples/treasure-hunt/app.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        with open(output_file_path, 'r') as output_file:
+            summary = output_file.read()
+
+        # Return clues
+        return summary
+    except Exception as e:
+        return jsonify({'error': str(e)}), 500
+
+if __name__ == '__main__':
+    app.run(debug=False)
```
Comment (cloudnautique) [inline examples/treasure-hunt/app.py:51]: Should the debug flag just be set with the env var in the Dockerfile? That way someone could still do local dev?

Comment (techmaharaj) [reply examples/treasure-hunt/app.py]: Fixed it by adding an env variable to Dockerfile and taking that in app.py.

File: examples/treasure-hunt/templates/index.html (hunk 3 above)
Comment (techmaharaj) [reply examples/treasure-hunt/templates/index.html]: Added a link to gptscript.ai

**Comment by techmaharaj** [comment]
Hi @cloudnautique I've fixed all the comments, please check.

**Review by cloudnautique — APPROVED** [review]

**Pull request merged**